node/service/src/lib.rs

```diff
@@ -1005,37 +1005,50 @@ pub fn build_full(
 	grandpa_pause: Option<(u32, u32)>,
 	jaeger_agent: Option<std::net::SocketAddr>,
 ) -> Result<NewFull<Client>, Error> {
+	let isolation_strategy = {
+		#[cfg(not(any(target_os = "android", target_os = "unknown")))]
+		{
+			let cache_path = config.database.path();
```

Comment:
This is basically the same approach we've taken in the av-store.

Reply:
This is actually the parent directory of the cache path, right? Maybe worth renaming it to `cache_base_path` or something like that.

Reply:
this is actually a good idea, thanks